### PR TITLE
Add tmpfs-enabled mssql images

### PR DESCRIPTION
**Pull request by pimeys**
The c-file is taken from https://github.com/t-oster/mssql-docker-zfs and it has GPLv3 license which is included.

### CODE REVIEW

File: .buildkite/pipeline.sh [hunk 1]
```diff
@@ -24,4 +24,8 @@ echo "steps:
   - label: \":rust: Release image\"
     command: cd release && make build && make push
     branches: master
+
+  - label: \":rust: Database images\"
```
Comment (dpetrick) [inline .buildkite/pipeline.sh:28]: These are not database images, but mssql images. misleading naming.

**Review by dpetrick — APPROVED** [review]
No idea what is going on here. From experience I can say that ramdisks have caused far more trouble than they're worth. We can try one more time.

**Comment by pimeys** [comment]
Let me explain what's going on here.

MSSQL will run just much better on ramdisk, leading to less deadlocks. It although needs one function from the filesystem that the tmpfs doesn't give, so we basically patch a different `open` to the image, allowing us to use other filesystems than extfs. If this goes bad, we need to backtrack and just wait for 40-50 minutes for MSSQL tests to finish single-threaded.

File: .buildkite/pipeline.sh (hunk 1 above)
Comment (pimeys) [reply .buildkite/pipeline.sh]: fixed

**Pull request merged**